subprojects/core/src/testFixtures/groovy/org/gradle/api/internal/file/TestFiles.java

```diff
@@ -96,4 +97,8 @@ public static JavaExecHandleFactory javaExecHandleFactory(File baseDir) {
     public static Factory<PatternSet> getPatternSetFactory() {
         return resolver().getPatternSetFactory();
     }
+
+    public static String systemSpecificAbsolutePath(String path) {
+        return OperatingSystem.current().isWindows()? "C:/" + path : "/" + path;
```

Comment:
So in many of our tests involving `JavaForkOptions`, we use `TestFiles.resolver()` which is a `FileResolver` without a base directory, which means any calls to `resolve()` have to use an absolute path.  In all of the cases where we use it, it doesn't have to be an existing path, just a resolvable path.  This was just a simple way to create a resolvable system-independent absolute path for those tests.  There are probably other solutions, too, I'm sure.